[Ramu Lakshmanan] iP

laksherthanlife opened the pull request:
# Quacky
> Feelin Stuck. Talk to the duck

Quacky makes your life so much easier to manage. It's

- text-based
- easy to learn
- ~~FAST~~ **SUPER FAST** to use

All you need to do is,
1. download it from [here](https://github.com/R-Laksh/ip/releases/tag/v0.1)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `run` method:
```
    public void run() {
        ui.showGreeting();
        while (ui.isRunning) {
            try {
                String command = ui.readCommand();
                Parser.parseCommand(command, tasks, ui);
            } catch (QuackyException e) {
                ui.showErrorMessage(e);
            }
        }
        try {
            storage.save(tasks);
        } catch (Storage.StorageOperationException e) {
            ui.showErrorMessage(e);
        }
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

laksherthanlife committed:
implemented level 1

laksherthanlife committed:
Add,List

laksherthanlife committed:
fixed formatting

laksherthanlife committed:
added TaskList class

laksherthanlife committed:
implemented v3.0

laksherthanlife committed:
added Deadline, Event, and Todo

laksherthanlife committed:
implemented v4.0

laksherthanlife committed:
implemented A-TextUiTesting

laksherthanlife committed:
implemented v5.0

laksherthanlife committed:
implemented v5.0

laksherthanlife committed:
Add ability to save tasks

laksherthanlife committed:
Add a read feature for Todos

The commit adds the ability to recover saved todo lists from past runs
to ensure that users can keep track of tasks

Achieved by adding a method to TaskList class to read the saved txt file
and parse it to recover todos

laksherthanlife committed:
Add LocalDate for tasks

Local date enables a more accurate representation of time for tasks and
minimises unexpected behaviour compared to using a string. The Deadline
and Event classes now take a LocalTime object in their constructor

laksherthanlife committed:
Add Implementation for UI and Parser classes

Current implementation of Quacky does not abide by the single responsibility principle. Hence the parsing logic is abstracted to the Parser class and the printed messages are abstracted to the UI class

laksherthanlife committed:
Implement Storage class
Currently Tasklist also has the behaviour of saving and loading. Storage class is
implemented to handle file reading and writing for previous runs

laksherthanlife committed:
Organise into Package

laksherthanlife committed:
Merge branch 'add-gradle-support'

laksherthanlife committed:
Add JUnit Tests

laksherthanlife committed:
Add javadocs for Quacky and Task class

laksherthanlife committed:
Add javadocs for all methods

laksherthanlife committed:
Follow Coding Standard

laksherthanlife committed:
Add find feature

laksherthanlife committed:
fixed merge conflicts

laksherthanlife committed:
merge find into main

kwuunnn commented on src/main/java/Quacky/Parser.java:
```diff
@@ -0,0 +1,107 @@
+        String[] keywords = command.split(" ", 2);
+        String commandWord = keywords[0];
+        switch (commandWord.toLowerCase()) {
+            case "list": {
```
Should there be indentation for `case` clauses?

kwuunnn commented on src/main/java/Quacky/Parser.java:
```diff
@@ -0,0 +1,107 @@
+            }
+            case "find": {
+                String keyword = keywords[1];
+                TaskList foundTasks = tasks.findTasksByKeyword(keyword);
```
Perhaps `tasksFound` would sound more like a noun for this variable? To obey code quality guidelines.

kwuunnn commented on src/main/java/quacky/Deadline.java:
```diff
@@ -0,0 +1,35 @@
+ * to be completed before a specific date.
+ */
+public class Deadline extends Task{
+    protected LocalDate by;
```
Perhaps `byDate` would sound more like a noun for this variable? To obey code quality guidelines.

kwuunnn commented on src/main/java/quacky/Event.java:
```diff
@@ -0,0 +1,40 @@
+ * that occurs over a period of time, from a start date to an end date.
+ */
+public class Event extends Task{
+    protected LocalDate from;
```
Perhaps `fromDate` would sound more like a noun for this variable? To obey code quality guidelines.

kwuunnn commented on src/main/java/quacky/Event.java:
```diff
@@ -0,0 +1,40 @@
+public class Event extends Task{
+    protected LocalDate from;
+    protected String stringFrom;
+    protected LocalDate to;
```
Perhaps `toDate` would sound more like a noun for this variable? To obey code quality guidelines.

kwuunnn reviewed (commented):
LGTM. 💯 👍 ❤️  

Just a few nits to fix.

laksherthanlife committed:
Fix deleted TaskList class

laksherthanlife committed:
Add checkstyle

laksherthanlife committed:
Add basic GUI

laksherthanlife committed:
Allow GUI to save tasks

naozee commented on src/main/java/quacky/Storage.java:
```diff
@@ -0,0 +1,133 @@
+    }
+
+    /**
+     * Returns true if the given path is acceptable as a storage file.
```
Might want to use @return when explaining what the function returns.

naozee commented on src/main/java/Quacky/Parser.java:
```diff
@@ -0,0 +1,95 @@
+        case "list": {
+            return ui.showList(tasks);
+        }
+        case "find": {
```
Might not want to use { in case? Not too sure but coding standard emits it.

naozee reviewed (commented):
LGTM apart from a few possible coding violations and comments maybe a little lacking? Good job!

laksherthanlife committed:
Add assertions

laksherthanlife committed:
Merge pull request #1 from R-Laksh/A-Assertions

Add assertions

laksherthanlife committed:
Improve naming quality of variables

Currently, variables such as from and by in the Event class are not in
noun form.

This needs to change as it is in violation of the style guide.

Lets rename variables to be in noun form to ensure code quality

laksherthanlife committed:
Remove deprecated methods in TaskList Class

Currently, the process of saving and storing the task list has been
abstracted to the Storage class. However, the methods for both can still
be found in the TaskList class.

This is a problem as it violates the Single Responsibility Principle and
creates confusion for people reviewing the code.

Lets address this by deleting both methods from the TaskList class
directly.

laksherthanlife committed:
Refactor parseCommand method in Parser class

Currently, the parseCommand method is very lengthy as it handles the
logic for each possible case.

This should be avoided as it makes the method too long, in violation of
the style guide.

Lets change the parser class so that it has separate methods to handle
each type of command word, which parseCommand invokes.

laksherthanlife committed:
Merge pull request #2 from R-Laksh/A-CodeQuality

Improve code quality

laksherthanlife committed:
Implement B-DetectAnomalies

laksherthanlife committed:
Merge pull request #3 from R-Laksh/branch-BCD-Extension

Implement B-DetectAnomalies

laksherthanlife committed:
Add handling for wrong date formats

laksherthanlife committed:
Add JavaDocs to Parser Class

laksherthanlife committed:
Add JUnit tests for Parser

laksherthanlife committed:
Fix bugs for delete command

laksherthanlife committed:
Make finishing touches

laksherthanlife committed:
Update README.md

laksherthanlife committed:
Add more testing

laksherthanlife committed:
Merge branch 'master' of https://github.com/R-Laksh/ip
